CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -471,35 +472,84 @@ def add_item(self, conditional_expression=None) -> DB_Response:
     def get_fridge_locations(self):
         pass
 
+    def update_fridge_report(self, fridgeId: str, fridge_report: dict) -> DB_Response:
+        """
+        Updates latestFridgeReport field with new Fridge Report.
+        This function is called when a new FridgeReport is added to the database
+        """
+        db_reponse = self.get_item(fridgeId=fridgeId)
+        if not db_reponse.is_successful():
+            # Fridge was not found
+            return db_reponse
+        fridge_dict = json.loads(db_reponse.json_data)
+        fridge_dict["latestFridgeReport"] = fridge_report
+        fridge_json_data = json.dumps(fridge_dict)
+        latestFridgeReport = json.dumps(fridge_report)
+        self.db_client.update_item(
+            TableName=self.TABLE_NAME,
+            Key={"id": {"S": fridgeId}},
+            ExpressionAttributeNames={"#LFR": "latestFridgeReport", "#JD": "json_data"},
+            ExpressionAttributeValues={
+                ":fr": {"S": latestFridgeReport},
+                ":fj": {"S": fridge_json_data},
+            },
+            UpdateExpression="SET #JD = :fj, #LFR = :fr",
```

Comment:
nit: can you explain this expression?

Reply:
Yeah they make it really confusing how you have to do this. The goal is to get here:
`SET json_data = {"S": fridge_json_data}, "#LFR": {"S": latestFridgeReport}`

but to get there we need to write something for **ExpressionAttributeNames** and **ExpressionAttributeValues**

------
Here's my best attempt at explaining it, lmk if its good and I'll add it in as a comment:
```
# JD and #LFR are mapped to the values set in ExpressionAttributeNames
# :fj and :fr are mapped to the values set in ExpressionAttributeValues
UpdateExpression becomes: "json_data":  {"S": fridge_json_data}, "latestFridgeReport": {"S": latestFridgeReport}
```

Reply:
done: 03ae4a50c0605ce95deb8658d6baf5a9a4183e16